domains/client/domain-operator/src/tests.rs

```diff
@@ -1494,6 +1494,260 @@ async fn test_false_invalid_bundles_inherent_extrinsic_proof_creation_and_verifi
     assert!(!ferdie.does_receipt_exist(bad_receipt_hash).unwrap());
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_true_invalid_bundles_undecodeable_tx_proof_creation_and_verification() {
+    let directory = TempDir::new().expect("Must be able to create temporary directory");
+
+    let mut builder = sc_cli::LoggerBuilder::new("");
+    builder.with_colors(false);
+    let _ = builder.init();
+
+    let tokio_handle = tokio::runtime::Handle::current();
+
+    // Start Ferdie
+    let mut ferdie = MockConsensusNode::run(
+        tokio_handle.clone(),
+        Ferdie,
+        BasePath::new(directory.path().join("ferdie")),
+    );
+
+    // Run Alice (a evm domain authority node)
+    let mut alice = domain_test_service::DomainNodeBuilder::new(
+        tokio_handle.clone(),
+        BasePath::new(directory.path().join("alice")),
+    )
+    .build_evm_node(Role::Authority, Alice, &mut ferdie)
+    .await;
+
+    let bundle_to_tx = |opaque_bundle| {
+        subspace_test_runtime::UncheckedExtrinsic::new_unsigned(
+            pallet_domains::Call::submit_bundle { opaque_bundle }.into(),
+        )
+        .into()
+    };
+
+    let undecodable_tx = || {
+        // 254 is reserved for tests like this, because some extrinsic codes will accept any data.
+        let undecodable_extrinsic = vec![254, 253, 252, 251, 250];
```

Comment:
Perhaps got collision by coincidence

Reply:
> Perhaps got collision by coincidence

Not sure follow what you mean here.

I'm mostly confused as to why this test inavlid extrinsic needs to be formatted like that. Scale decoding should fail irrespective as long as it some random data. Sometimes it might have a collision of valid pallet ID but rest of the data should be invalid leading to an same error anyway

Reply:
I think extrinsics are encoded using the extrinsic number as a prefix, along with the length?

So if we happen to get a real extrinsic number at random which accepts any length and data bytes, and the random length is less than the length of the random data, the decoding will pass when it shouldn't.

Happy to try it out, but it seems like a reliable test is better?